### PR TITLE
Add bit or/and/xor function

### PR DESCRIPTION
**Pull request by ouyuanning**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #3619

## What this PR does / why we need it:
Add bit or/and/xor function support.
(they are bit operator, not bool, not agg)

### CODE REVIEW

File: pkg/sql/plan/function/operator/op_bit.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	"github.com/matrixorigin/matrixone/pkg/container/nulls"
+	"github.com/matrixorigin/matrixone/pkg/container/types"
+	"github.com/matrixorigin/matrixone/pkg/container/vector"
+	"github.com/matrixorigin/matrixone/pkg/encoding"
+	"github.com/matrixorigin/matrixone/pkg/vm/process"
+	"golang.org/x/exp/constraints"
+)
+
+type opBitT interface {
+	constraints.Unsigned
```
Comment (fengttt) [inline pkg/sql/plan/function/operator/op_bit.go:27]: Why you don't want to include signed?

File: pkg/sql/plan/function/operator/op_bit.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+func OpBitOrFun[T opBitT](args []*vector.Vector, proc *process.Process) (*vector.Vector, error) {
+	return OpBitBase(args, proc, args[0].GetType(), opBitOr[T])
+}
+
+func OpBitXorFun[T opBitT](args []*vector.Vector, proc *process.Process) (*vector.Vector, error) {
+	return OpBitBase(args, proc, args[0].GetType(), opBitXor[T])
+}
+
+func OpBitBase[T opBitT](vectors []*vector.Vector, proc *process.Process, typ types.Type, afn opBitFun[T]) (*vector.Vector, error) {
+	left, right := vectors[0], vectors[1]
```
Comment (fengttt) [inline pkg/sql/plan/function/operator/op_bit.go:57]: Can this OpBitBase reused other operators?   Basically, what is the difference between bit_and/or with add?   You get two number and compute another.

Comment (ouyuanning) [reply pkg/sql/plan/function/operator/op_bit.go:57]: yes.   bit and/or/xor will use OpBitBase.

File: pkg/sql/plan/function/operator/op_bit.go (hunk 1 above)
Comment (ouyuanning) [reply pkg/sql/plan/function/operator/op_bit.go:27]: I register & | ^ for constraints.Unsigned  only.

then  constraints.signed will cast to unsigned.  if constraints.signed is negative， cast will throw error.

I don't know if it's a good idea

File: pkg/sql/plan/function/operator/op_bit.go (hunk 2 above)
Comment (ouyuanning) [reply pkg/sql/plan/function/operator/op_bit.go:57]: only one difference with add/multiplication/div...
the result of bit operator will not check overflow

**Review by qingxinhome — APPROVED** [review]

**Review by aressu1985 — APPROVED** [review]

**Review by aunjgr — APPROVED** [review]

**Pull request merged**